app/infrastructure/db/repositories/sqlalchemy_user_repository.py

```
@@ -96,3 +99,27 @@ def update_password(self, user_id: str, password_hash: str) -> User | None:
+            self._db.query(UserORM)
+            .filter(UserORM.provisioning_token == token)
+            .one_or_none()
+        )
+        if row is None:
+            return None
+        return self._to_domain(row)
+
+    def set_provisioning_token(self, user_id: str, token: str) -> None:
+        row = self._db.get(UserORM, user_id)
+        if row is None:
+            raise ValueError(f"User {user_id} not found")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)